### PR TITLE
[docs] Refactor the rest of the 'Test API' section

### PR DESCRIPTION
**Pull request by VasilyStrelyaev**

### CODE REVIEW

File: docs/articles/documentation/concepts/built-in-wait-mechanisms.md [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
+title: Built-In Wait Mechanisms 
+permalink: /documentation/concepts/built-in-wait-mechanisms.html
 redirect_from:
+  - /documentation/test-api/built-in-waiting-mechanisms.html
   - /documentation/test-api/waiting-for-page-elements-to-appear.html
 ---
-# Built-In Waiting Mechanisms
+# Built-In Wait Mechanisms
 
 TestCafe has built-in automatic waiting mechanism and does not require dedicated API to wait for redirects or page elements to appear.
```
Comment (arminal) [inline docs/articles/documentation/concepts/built-in-wait-mechanisms.md:11]:
```suggestion
TestCafe has a built-in automatic waiting mechanism and does not require dedicated API to wait for redirects or page elements to appear.
```

File: docs/articles/documentation/concepts/typescript-and-coffescript.md [hunk 2]
```diff
@@ -14,9 +18,9 @@ painless scalability, check-as-you-type code verification and much more.
 
 TestCafe bundles the TypeScript declaration file with the npm package, so you do not need to install it separately.
 
-## Writing Tests with TypeScript
+### Write Tests with TypeScript
 
-To start writing tests with TypeScript, install TestCafe into your project directory. For more information, see [Installing TestCafe](../using-testcafe/installing-testcafe.md#local-installation).
+To start writing tests with TypeScript, install TestCafe into your project directory. For more information, see [Install TestCafe](../guides/basic-guides/install-testcafe.md#local-installation).
 
 When writing test in TypeScript, it is required to import TestCafe module first.
```
Comment (arminal) [inline docs/articles/documentation/concepts/typescript-and-coffescript.md:25]:
```suggestion
When writing test in TypeScript, you must import the TestCafe module first.
```

File: docs/articles/documentation/concepts/typescript-and-coffescript.md [hunk 3]
```diff
@@ -29,37 +33,37 @@ file and will show you code completion hints for TestCafe API:
 
 ![Writing Tests with TypeScript](../../images/typescript-support.png)
 
-> If installed [globally](../using-testcafe/installing-testcafe.md#global-installation), TestCafe will successfully compile and run your tests written in TypeScript.
+> If installed [globally](../guides/basic-guides/install-testcafe.md#global-installation), TestCafe will successfully compile and run your tests written in TypeScript.
 In this case, your IDE will not be able to find the TestCafe declaration file and provide code completion.
 
 Now, you can write tests in the same manner as in JavaScript.
 When you run a test Testcafe will output if there are any compilation errors.
```
Comment (arminal) [inline docs/articles/documentation/concepts/typescript-and-coffescript.md:40]:
```suggestion
When you run a test, Testcafe will output if there are any compilation errors.
```
will output what?

File: docs/articles/documentation/concepts/typescript-and-coffescript.md [hunk 4]
```diff
@@ -29,37 +33,37 @@ file and will show you code completion hints for TestCafe API:
...
-> [Custom Properties](./selecting-page-elements/selectors/extending-selectors.md#custom-properties)
-> and [Custom Methods](./selecting-page-elements/selectors/extending-selectors.md#custom-methods)
+> [Custom Properties](../guides/basic-guides/select-page-elements.md#custom-properties)
+> and [Custom Methods](../guides/basic-guides/select-page-elements.md#custom-methods)
 > sections to learn how to extend selectors in TypeScript.
 
-## Customize Compiler Options
+### Customize Compiler Options
 
 TestCafe allows you to specify [TypeScript compiler options](https://www.typescriptlang.org/docs/handbook/compiler-options.html) in the `tsconfig.json` file. You can use these options to enable JSX compilation, import code or typings with `paths` aliases, set aliases to React typings, or customize other compiler settings.
```
Comment (arminal) [inline docs/articles/documentation/concepts/typescript-and-coffescript.md:50]:
```suggestion
TestCafe allows you to specify [TypeScript compiler options](https://www.typescriptlang.org/docs/handbook/compiler-options.html) in the `tsconfig.json` file. You can use these options to enable the JSX compilation, import code or typings with `paths` aliases, set aliases to React typings, or customize other compiler settings.
```
is typings a real word?

File: docs/articles/documentation/guides/basic-guides/debug.md [hunk 5]
```diff
@@ -1,43 +1,35 @@
...
 
-* [Debugging Test Code](#debugging-test-code)
+* [Debug Test Code](#debug-test-code)
 * [Client-Side Debugging](#client-side-debugging)
 * [Options Useful for Debugging](#options-useful-for-debugging)
 
-## Debugging Test Code
+## Debug Test Code
 
 You can debug test code in Chrome Developer Tools and popular IDEs. See the following recipes for details.
```
Comment (arminal) [inline docs/articles/documentation/guides/basic-guides/debug.md:16]: recipes?

File: docs/articles/documentation/guides/basic-guides/debug.md [hunk 6]
```diff
@@ -1,43 +1,35 @@
...
 
-* [Debug in Chrome Developer Tools](../recipes/debug-in-chrome-dev-tools.md)
-* [Debug in Visual Studio Code](../recipes/debug-in-visual-studio-code.md)
+* [Debug in Chrome Developer Tools](../../recipes/debug-in-chrome-dev-tools.md)
+* [Debug in Visual Studio Code](../../recipes/debug-in-visual-studio-code.md)
 
 ## Client-Side Debugging
 
-TestCafe provides the `t.debug` method that pauses the test and allows you to debug using the browser's developer tools.
+TestCafe provides the [t.debug](../../reference/test-api/testcontroller/debug.md) method that pauses the test and allows you to debug using the browser's developer tools.
```
Comment (arminal) [inline docs/articles/documentation/guides/basic-guides/debug.md:23]: allows you to debug what?

File: docs/articles/documentation/guides/basic-guides/debug.md (hunk 5 above)
Comment (VasilyStrelyaev) [reply docs/articles/documentation/guides/basic-guides/debug.md:16]: "Recipes" or, collectively, "Cookbook" is a common term for documents that show specific scenarios in depth.

UI frameworks like `react` or `vue` have such sections:
https://reactjs.org/docs/testing-recipes.html
https://vuejs.org/v2/cookbook/

Most tools I use in my work, like site generators, also do:
https://www.gatsbyjs.org/docs/recipes/

File: docs/articles/documentation/guides/basic-guides/debug.md (hunk 6 above)
Comment (VasilyStrelyaev) [reply docs/articles/documentation/guides/basic-guides/debug.md:23]: Oh, yes, it should read:

_TestCafe provides the t.debug method that pauses the test and allows you to debug it with the browser's developer tools._

Comment (arminal) [reply docs/articles/documentation/guides/basic-guides/debug.md:23]: great

**Pull request merged**